src/lib/rest/httpHeaderAdd.cpp

```diff
@@ -92,7 +92,16 @@ void httpHeaderLinkAdd(const char* _url)
   }
 
   urlLen = strlen(url);
-  if (urlLen > sizeof(link) + LINK_REL_AND_TYPE_SIZE + 5)
+
+  // make sure we don't sprintf() more than INT_MAX chars
+  // (which is what sprintf() can handle)
+  if (urlLen + LINK_REL_AND_TYPE_SIZE + 5 > INT_MAX)
```

Comment:
Here.
Let's change the sprintf to snprintf (even though the logic before this call assures there's room enough in the buffer).
And, create a local variable for the total size needed (urlLen + LINK_REL_AND_TYPE_SIZE + 5), to not do the addition twice.

I truly believe we can avoid the comparison with INT_MAX, which makes very little sense (does your compiler really complain if we don't compare with INT_MAX?)

The idea is to make these few lines a bit simpler.
If we don't do that, I'll end up doing it myself and I might then break your cross-compilation.
So, better to do it together.